File: Pjesa_0/oscillator_prod_python/scripts/plot_results.py
#!/usr/bin/env python3
"""Lexon CSV (t,x,v) dhe vizaton x(t) dhe energjinë."""
from __future__ import annotations
import argparse
import numpy as np
import matplotlib.pyplot as plt
from oscillator.models import OscillatorParams, energy

def main():
    ap = argparse.ArgumentParser()
    ap.add_argument("--csv", required=True)
    ap.add_argument("--omega", type=float, default=2.0, help="duhet për energjinë")
    args = ap.parse_args()

    data = np.loadtxt(args.csv, delimiter=",", skiprows=1)
    t, x, v = data[:,0], data[:,1], data[:,2]
    p = OscillatorParams(omega=args.omega)
    E = np.array([energy(np.array([x[i], v[i]]), p) for i in range(len(t))])

    plt.figure()
    plt.plot(t, x, label="x(t)")
    plt.xlabel("t")
    plt.ylabel("x")
    plt.legend()
    plt.show()

    plt.figure()
    plt.plot(t, E - E[0], label="E - E0")
    plt.xlabel("t")
    plt.ylabel("Devijimi i energjisë")
    plt.legend()
    plt.show()

if __name__ == "__main__":
    main()

File: Pjesa_0/oscillator_prod_julia/scripts/plot_results.jl
using ArgParse
using DelimitedFiles
using Plots
using OscillatorProdDemo

function build_parser()
    s = ArgParseSettings()
    @add_arg_table s begin
        "--csv"
            help = "skedari CSV (t,x,v)"
            arg_type = String
            required = true
        "--omega"
            help = "ω (për energjinë)"
            arg_type = Float64
            default = 2.0
    end
    return s
end

function main()
    args = parse_args(build_parser())
    lines = readlines(args["csv"])
    data = readdlm(IOBuffer(join(lines[2:end], "\n")), ',', Float64)
    t = data[:,1]; x = data[:,2]; v = data[:,3]
    p = OscillatorParams(args["omega"])
    E = [energy((x[i], v[i]), p) for i in eachindex(t)]

    plot(t, x, xlabel="t", ylabel="x", label="x(t)")
    display(current())

    plot(t, E .- E[1], xlabel="t", ylabel="E - E0", label="Devijimi i energjisë")
    display(current())
end

main()
